### ISSUE TITLE
[BUG] Can't drop privilege as nonroot user

### ISSUE THREAD

**Issue by Itzdlg** [docker-mailserver/docker-mailserver#2627]
### Miscellaneous first checks

- [X] I checked that all ports are open and not blocked by my ISP / hosting provider.
- [X] I know that SSL errors are likely the result of a wrong setup on the user side and not caused by DMS itself. I'm confident my setup is correct.

### Affected Component(s)

Unable to start the container with a non-root user.

### What happened and when does this occur?

```Markdown
I'm using docker-compose to launch docker-mailserver, but I've recently went on the hunt to properly manage the permissions of all the services on the container. In order to run the container, from the root user I run `sudo -u mailserver docker-compose up`. This works fine if I don't specify a user ID in the docker-compose.yml file, but once I do, supervisord complains: "Can't drop privilege as nonroot user."
```

### What did you expect to happen?

```Markdown
I expect for the container to run properly even without root privileges, assuming the user provided has rights for all the resources on the file system.
```

### How do we replicate the issue?

```Markdown
1. Set a string user ID in the docker-compose under `services.mailserver.user`
2. Attempt to run `docker-compose up` normally

or supply a user ID with the docker command.
```

### DMS version

docker.io/mailserver/docker-mailserver:latest

### What operating system is DMS running on?

Linux

### What instruction set architecture is DMS running on?

x86_64 / AMD64

### What container orchestration tool are you using?

Docker Compose

### docker-compose.yml

```yml
version: "3"
services:
  mailserver:
    image: docker.io/mailserver/docker-mailserver:latest
    container_name: mailserver
    # If the FQDN for your mail-server is only two labels (eg: example.com),
    # you can assign this entirely to `hostname` and remove `domainname`.
    hostname: mail
    domainname: schooltests.me
    env_file: mailserver.env
    # More information about the mail-server ports:
    # https://docker-mailserver.github.io/docker-mailserver/edge/config/security/understanding-the-ports/
    # To avoid conflicts with yaml base-60 float, DO NOT remove the quotation marks.
    ports:
      - "25:25"    # SMTP  (explicit TLS => STARTTLS)
      - "143:143"  # IMAP4 (explicit TLS => STARTTLS)
      - "465:465"  # ESMTP (implicit TLS)
      - "587:587"  # ESMTP (explicit TLS => STARTTLS)
      - "993:993"  # IMAP4 (implicit TLS)
    volumes:
      - /home/mailserver/mail-data/:/var/mail/
      - /home/mailserver/mail-state/:/var/mail-state/
      - /home/mailserver/mail-logs/:/var/log/mail/
      - /home/mailserver/config/:/tmp/docker-mailserver/
      - /etc/localtime:/etc/localtime:ro
      - "/etc/ssl/certs/schooltests.me/:/var/ssl/"
    user: "1000"
    restart: always
    stop_grace_period: 1m
    cap_add:
      - NET_ADMIN
      - SYS_PTRACE
```

### Relevant log output

```bash
mailserver    | Error: Can't drop privilege as nonroot user
mailserver    | For help, use /usr/bin/supervisord -h
```

### What level of experience do you have with Docker and mail servers?

- [ ] I am inexperienced with docker
- [ ] I am inexperienced with mail servers
- [ ] I am uncomfortable with the CLI

### Code of conduct

- [X] I have read this project's [Code of Conduct](https://github.com/docker-mailserver/docker-mailserver/blob/master/CODE_OF_CONDUCT.md) and I agree
- [X] I have read the [README](https://github.com/docker-mailserver/docker-mailserver/blob/master/README.md) and the [documentation](https://docker-mailserver.github.io/docker-mailserver/edge/) and I searched the [issue tracker](https://github.com/docker-mailserver/docker-mailserver/issues?q=is%3Aissue) but could not find a solution

**Comment by polarathene** [comment]
Not a bug. This is a feature request.

If you want to implement support for that, you're welcome to attempt it. Otherwise raise a feature request if you like, but it's unlikely someone else will handle this for you unless they are motivated with their own interest for that, and the feature request will go stale and be closed.

There is only a few volunteers that work on this project, while we'd love to support many requests we do need to keep within the scope of what we have time to handle.

---

That said, some of your concerns may be addressable:

- For volume bind mounts, you can better manage ownership of files written on the host with IDMAPPED mounts when that feature is supported by your kernel, filesystem and container runtime.
- Docker and Podman have specific rootless modes. I have no experience with these but you can find both projects document such. Our own docs have received a contribution [detailing rootless container setup with Podman](https://docker-mailserver.github.io/docker-mailserver/edge/config/advanced/podman/). This will run the container externally as a non-root user AFAIK, so the containers internal root user has reduced risk of damage from an attacker breaking out of it.

Also, to my knowledge setting an override to the container user via the `user` option isn't advisable for this. Often a docker image setup with a root user can expect to be run as root and have access to read/write locations restricted to root. When you override the user to run as, you prevent that access. I know this from attempting the same in other containers (one with nginx IIRC).

You need to properly modify the image. I believe your main concern is not what is going on within the container, but the risk to the host system. Many articles you may have come across that warn about that are often before rootless containers were supported well, and many of the earlier vulnerabilities since addressed.

**Comment by Itzdlg** [comment]
I see, then. I thought this would have been unintended as I haven't tried this with any other images/applications yet. I hadn't read any articles about this, instead I was just trying to follow (what I thought to be) best practices for isolating each service. However, I understand that implementing this is definitely not a priority nor a small undertaking, and well I don't have the need or knowledge to do it myself.

I'll be closing this issue as it's not categorized properly, and if anyone else wants this they should open a feature request. Thank you polarathene for the quick triage & all of the other contributors for their work into the mail server!
